### PR TITLE
Yolo/use yamllint

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
Add `yamllint` configuration file, to enforce a style on `yaml` files

### CODE REVIEW

File: .yamllint.yml
Comment (touilleMan) [inline .yamllint.yml:1]: What's the reason we don't use yamllint's default configuration ?

File: .landscape.yml
Comment (touilleMan) [inline .landscape.yml:13]: @vxgmichel do you know if this config file is still used ? I have the feeling we don't use this service since ages...

File: .pre-commit-config.yaml
Comment (touilleMan) [inline .pre-commit-config.yaml:2]: I guess we should add yamllint to pre-commit config

This should be fairly simple given yamllint repo contains a [pre-commit config](https://github.com/adrienverge/yamllint/blob/master/.pre-commit-hooks.yaml)
```yaml
- repo: https://github.com/adrienverge/yamllint
  rev: v1.26.3
  hooks:
  - id: yamllint
```

**Review by touilleMan — CHANGES_REQUESTED** [review]

**Comment by touilleMan** [comment]
I'd rather wait for #2371 to be merged before merging this one (given azure-pipeline.yml and snapcraft.yml gets modified in this PR)

File: .pre-commit-config.yaml
Comment (FirelightFlagboy) [reply .pre-commit-config.yaml:2]: I've planned that in another PR

File: .yamllint.yml
Comment (FirelightFlagboy) [reply .yamllint.yml:1]: 1. I find the default configuration a little to permissive for braces and bracket
2. I prefer the config to be explicite

File: .landscape.yml
Comment (vxgmichel) [reply .landscape.yml:13]: The commit is from 6 years ago, I wasn't event here :)

Comment (FirelightFlagboy) [reply .landscape.yml:13]: the file is removed

File: .yamllint.yml
Comment (touilleMan) [reply .yamllint.yml:1]: I'd rather have one less configuration file to worry about

> I prefer the config to be explicit

I disagree: it's better to have one way to do the formatting and don't think about it (see for instance black for Python or rust-fmt compared to the mad house that is C formatting)

> I find the default configuration a little to permissive for braces and bracket

See my previous point ;-)

**Review by touilleMan — APPROVED** [review]

**Pull request merged**
